logging/cloud-client/snippets.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This application demonstrates how to perform basic operations on logs and
+log entries with Stackdriver Logging.
+
+For more information, see the README.md under /logging and the
+documentation at https://cloud.google.com/logging/docs.
+"""
+
+import argparse
+
+from gcloud import logging
+
+
+def write_entry(logger_name):
+    """Writes log entries to the given logger."""
+    logging_client = logging.Client()
+
+    # This log can be found in the Cloud Logging console under 'Custom Logs'.
+    logger = logging_client.logger(logger_name)
+
+    # Make a simple text log
+    logger.log_text('Hello, world!')
+
+    # Simple text log with severity.
+    logger.log_text('Goodbye, world!', severity='ERROR')
+
+    # Struct log. The struct can be any JSON-serializable dictionary.
+    logger.log_struct({
+        'name': 'King Arthur',
+        'quest': 'Find the Holy Grail',
+        'favorite_color': 'Blue'
+    })
+
+    print('Wrote logs to {}.'.format(logger.name))
+
+
+def list_entries(logger_name):
+    """Lists the most recent entries for a given logger."""
+    logging_client = logging.Client()
+    logger = logging_client.logger(logger_name)
+
+    print('Latest entries for logger {}:'.format(logger.name))
+
+    entries, token = logging_client.list_entries(
```

Comment:
Two things:

1) If you don't loop on the token, you're going to have a bad time. The API will return 0 entries at first then return 5 on the next one, the token is also used to indicate "loading"

2) You should just be able to do logger.list() now. There was a gcloud-python bug I fixed that should be released now)

https://github.com/GoogleCloudPlatform/gcloud-python/pull/1848